src/commands.cc

```diff
@@ -2535,6 +2536,24 @@ const CommandDesc change_directory_cmd = {
     }
 };
 
+const CommandDesc present_working_directory_cmd = {
+    "present-working-directory",
+    "pwd",
+    "present-working-directory: get the server's current working directory",
+    no_params,
+    CommandFlags::None,
+    CommandHelper{},
+    CommandCompleter{},
+    [](const ParametersParser&, Context& context, const ShellContext&)
+    {
+        char cwd[1024];
```

Comment:
Better use `PATH_MAX` here, 1024 seems short.

Reply:
Done.

I've also made some additional PATH_MAX related changes in rest of the code base. See https://github.com/mawww/kakoune/pull/4431/commits/76f925e9d4dfb974fcbd0c0a5299b1d1420e27b3 . Whether or not this PR gets accepted I think [that commit](https://github.com/mawww/kakoune/pull/4431/commits/76f925e9d4dfb974fcbd0c0a5299b1d1420e27b3 ) should be cherry-picked into the code? cc: @mawww

Reply:
@lenormf Thanks for your comments related to PATH_MAX on https://github.com/mawww/kakoune/commit/76f925e9d4dfb974fcbd0c0a5299b1d1420e27b3#r60092366 -- useful!

(I am shifting the conversation back here as it is easier to refer back to a conversation on a PR than a loose commit coming from my fork)

I had a glance at the link you provided -- https://stackoverflow.com/questions/9449241/where-is-path-max-defined-in-linux

The headline take away for me is that PATH_MAX is a dynamic system quantity that can vary. So for instance different file systems can have different values on Linux. Since kakoune can edit files on different file systems the effective PATH_MAX will vary. So its not really even possible to define something like this at compile time.

Regarding the kakoune code base. We have three scenarios currently on the master branch:
- Usage of PATH_MAX
- Usage of PATH_MAX+1
- Usage of hard coded number like 1024

I changed all of them to option 1 for consistency in https://github.com/mawww/kakoune/commit/76f925e9d4dfb974fcbd0c0a5299b1d1420e27b3 .

So if I understand correctly you are advocating using PATH_MAX+1 in case there is an OS that defines PATH_MAX to be non-inclusive of nul? Linux BTW very clearly defines that it includes `nul`.

PATH_MAX has the advantage that it looks cleaner in code. PATH_MAX+1 only has the minor advantage if:

(a) We are using an operating system that `#defines` PATH_MAX as _not_ including nul
_and_
(b) You have a file that is PATH_MAX characters long. In that case the last character will get truncated unnecessarily. (Anything greater than PATH_MAX will get truncated anyways) 

Not too say the whole concept of PATH_MAX is flawed anyways. I can live with both PATH_MAX and PATH_MAX+1.

---
So finally:
- I simply reverted my changes to other people's code. So the hardcoded 1024 stays and usages of PATH_MAX and PATH_MAX+1 stays. Making this consistent can come later.
- I changed my own use of PATH_MAX to PATH_MAX+1 for the `pwd` feature.

Reply:
Got it, I created #4434 for this specific topic.